### ISSUE TITLE
[BUG] Header click bug

### ISSUE THREAD

**Issue by misochenk0** [rupali-codes/LinksHub#2312]
### Description

When I click anywhere on the page header it redirects me to the main page
Maybe a better solution here would be preventing this behavior and redirecting users to the main page only after they click on logo?

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)

**Comment by rupali-codes** [comment]
Hey @misochenk0 thank you for noticing, as you've already assigned 2 issues, we're opening this issue for others. And, when you're done with the other assigned issue and this issue still remains unassigned then we'll assign it to you.

**Comment by anantgna** [comment]
@rupali-codes I can fix this out.

**Comment by Anmol-Baranwal** [comment]
@anantgna 
I have assigned you the issue. Please proceed with it, and don't hesitate to tag any of the maintainers if you need any help.

**Comment by anantgna** [comment]
Thanks @Anmol-Baranwal

**Comment by anantgna** [comment]
Hi @Anmol-Baranwal,

We've encountered an issue with our current implementation where the useOnClickOutside function is responsible for closing the side nav bar and redirecting to the main page. Currently, this function targets the **restrict-click-outside** class. I'm considering removing this class from unwanted components and specifically implementing it on the logo component. Do you think this approach is appropriate, or would you suggest another solution?

@rupali-codes @CBID2, I'd appreciate your inputs on this as well. Thanks!

**Comment by anantgna** [comment]
Sure @Anmol-Baranwal will take care of this.

**Comment by anantgna** [comment]
Hey @Anmol-Baranwal, we need to redirect to home screen only when user clicks on linkshub logo, else there should be no action right?
And side bar should also only be collapsed on click on logo, else it shouldn't close, am I right?

**Comment by Anmol-Baranwal** [comment]
> Hey @Anmol-Baranwal, we need to redirect to home screen only when user clicks on linkshub logo, else there should be no action right? And side bar should also only be collapsed on click on logo, else it shouldn't close, am I right?

Yes, only after clicking the logo, it should redirect to the state like the home page of [linkshub](https://www.linkshub.dev/), which you can check. All the categories in the sidebar are collapsed and so on.

**Comment by rupali-codes** [comment]
Closing issue due to inactveness.
